python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -785,6 +813,46 @@ def __construct__(self, args):
             from Ganga.GPIDev.Base.Proxy import TypeMismatchError
             raise TypeMismatchError("Constructor expected one or zero non-keyword arguments, got %i" % len(args))
 
+    @staticmethod
+    def _getChildLock(child_name, parent):
+        whole_path = GangaObject._full_childPath(child_name, parent)
+
+        top_parent = parent._getRoot()
+
+        top_parent_child_locks = top_parent._getChildLocks()
+
+        if whole_path not in top_parent_child_locks:
+            top_parent_child_locks[whole_path] = threading.Lock()
+
+        return top_parent_child_locks[whole_path]
+
+    @staticmethod
+    def _full_childPath(child_name, parent):
+        child_path = GangaObject._getChildPathStr(parent)
+        whole_path = child_path + "/" + child_name
+        return whole_path
+
+    @staticmethod
+    def _getChildPathStr(child):
+        names = []
+        names.append(_getName(child))
+        parent = child._getParent()
+
+        while parent is not None:
+            names.append(_getName(parent))
+            parent = parent._getParent()
+
+        child_str = "/".join(reversed(names))
+
+        return child_str
+
+    @staticmethod
+    def _getParentLock(child):
+        ## Top level parent
+        parent = child._getRoot()
+        parent_lock = parent._getLock()
+        return parent_lock
```

Comment:
This is really `_getRootLock`, right? We should make sure we're naming things appropriately to avoid confusion.